### ISSUE TITLE
[CORE] update template functions

### ISSUE THREAD

**Issue by remrem** [BoboTiG/blogotext#211]
[4.0] Relatif à l'issue #133 

Je bosse sur les functions et les process liés au template public, en vrac : 
- bascule sur de l'anglais (tags de template, comments, code...)
- réorganisation du code (éclatage pour lisibilité, ajout/modif/suppression de fonctions...)
- quelques petites optimisations
- séparation admin/public
- ajout de documentation https://github.com/BoboTiG/blogotext/wiki/template---tags
...

Mauvaises nouvelles (en vrac et non-exhaustif) : 
- pas de rétro compatibilité < 4.0 des thémes, rien que pour gérer les anciens+nouveaux tags c'est pas la peine
- le code / functions liés au template gonfle un peu (à voir si possibilité d'écrémage plus tard)

Bonnes nouvelles (en vrac et non-exhaustif) : 
- La création de template devrait être un poil plus facile (ajout de documentation, convention de nommage des tags...)
- Le core est plus facilement lisible, du coup, l'ajout de tags, de type de page devrait être plus facile
- Les nouveaux templates seront plus ouverts aux possibilités (langues, architecture HTML...)

Tant que j'y suis, des besoins (tag ...) ?

**Comment by B4rb3rouss** [comment]
> le code / functions liés au template gonfle un peu (à voir si possibilité d'écrémage plus tard)

Juste une idée comme ça, ça ne serait pas possible de faire un script de migration qui parcourrait la BDD  et ferait les substitutions qui vont bien à la sed : 
    
    sed s;ancien_tag;nouveau_tag;g

**Comment by remrem** [comment]
Oui, faisable. On peux même envisager d'inclure un "converter" dans le process de mise à jour 3.7 > 4.0. Par contre, je ne pense pas que la BDD soit touchée, je ne vois pas d'utilité à intégrer les tags de template dans un article/lien(...), je rate quelques chose ?

**Comment by B4rb3rouss** [comment]
oui, "BDD" était mal choisi, je pensais aux données tout simplement.
